pyproject.toml

```diff
@@ -213,11 +213,6 @@ ignore = [
     "B017",    #`pytest.raises(Exception)` should be considered evil
     "RUF005",  # Consider "...."instead of concatenation
     "RUF012",  # Mutable class attributes should be annotated with "typing.ClassVar"
-    "TC001",   # Type-checking block
-    "TC002",   # Type-checking block
-    "TC003",   # Type-checking block
-    "TC004",   # Type-checking block
```

Comment:
do we want to auto add future annotations? https://docs.astral.sh/ruff/settings/#lint_future-annotations

Reply:
I'm afraid to blindly activate that, because Pydantic `BaseModel`s would be affected. Those models require the referenced types during runtime mode, but `ruff` considers them only for type hinting. Activating that would review 2225 changes.

Reply:
Nevertheless, I've added [runtime-evaluated-base-classes](https://docs.astral.sh/ruff/settings/#lint_flake8-type-checking_runtime-evaluated-base-classes) to handle it automatically.